File: Python/8kyu/8kyu-quarter-of-the-year.py
# Quarter of the year
# Langugage: Python
# Version: 3.10
# Difficulty: 8 kyu

# Without using math
def quarter_of(month):
    return -(-month // 3)

# Using math
from math import ceil
def quarter_of(month):
    return ceil(month / 3)


# Tests
from solution import quarter_of
import codewars_test as test

def dotest(n, expected):
    actual = quarter_of(n)
    test.assert_equals(actual, expected, f"Test failed with month = {n}")
    
@test.describe("Tests")
def test_group():
    @test.it("Sample tests")
    def test_case():
        dotest(3, 1)
        dotest(8, 3)
        dotest(11, 4)

File: R/8kyu/8kyu-quarter-of-the-year.r
# Quarter of the year
# Langugage: R
# Version: 3.4.1
# Difficulty: 8 kyu

# Solution using dplyr
library(dplyr)
quarter_of <- function(month){
  case_when(between(month, 1, 3) ~ 1,
            between(month, 4, 6) ~ 2,
            between(month, 7, 9) ~ 3,
            between(month, 10, 12) ~ 4)
}

# Using base R ceiling function
quarter_of <- function(month){
   ceiling(month / 3)
}


# Tests
test_that('Basic tests', {
  expect_equal(quarter_of(3), 1)
  expect_equal(quarter_of(8), 3)
  expect_equal(quarter_of(11), 4)
})